app/src/main/java/com/example/moviews/screen/moviedetail/MovieDetailFragment.kt

```diff
@@ -2,21 +2,121 @@ package com.example.moviews.screen.moviedetail
 
 import com.example.moviews.R
 import com.example.moviews.base.BaseFragment
+import com.example.moviews.data.model.Cast
+import com.example.moviews.data.model.Company
+import com.example.moviews.data.model.Genre
+import com.example.moviews.data.model.Movie
+import com.example.moviews.repository.RepositoryUtils
+import com.example.moviews.screen.moviedetail.adapter.CastAdapter
+import com.example.moviews.screen.moviedetail.adapter.CompanyAdapter
+import com.example.moviews.screen.moviedetail.adapter.RecommendAdapter
+import com.example.moviews.screen.search.adapter.GenresAdapter
+import com.example.moviews.utils.Constant
+import com.example.moviews.utils.loadImage
+import com.example.moviews.utils.showToast
+import kotlinx.android.synthetic.main.fragment_movie_detail.*
+import kotlinx.android.synthetic.main.fragment_movie_detail.imageBack
+import java.lang.Exception
 
-class MovieDetailFragment(idMovie: Int) : BaseFragment() {
+class MovieDetailFragment(
+    private val idMovie: Int
+) : BaseFragment(), MovieDetailContract.View {
+
+    private var movie: Movie? = null
+    private val recommendAdapter = RecommendAdapter(this::onClickRecommend)
+    private val companyAdapter = CompanyAdapter(this::onClickCompanies)
+    private val castAdapter = CastAdapter(this::onClickCasts)
+    private val genresAdapter = GenresAdapter(this::onClickGenres)
+    private var presenter: MovieDetailPresenter? = null
     override val layoutID: Int
         get() = R.layout.fragment_movie_detail
 
     override fun initViews() {
+        recyclerGenres.apply {
+            setHasFixedSize(true)
+            adapter = genresAdapter
+        }
+        recyclerCast.apply {
+            setHasFixedSize(true)
+            adapter = castAdapter
+        }
+        recyclerCompany.apply {
+            setHasFixedSize(true)
+            adapter = companyAdapter
+        }
+        recyclerRecommend.apply {
+            setHasFixedSize(true)
```

Comment:
dòng này để làm gì

Reply:
gọi cái này để item không phải tính toán lại size nếu có thay đổi data, lướt mượt hơn ạ